app/scripts/controllers/preferences.js

```diff
@@ -642,43 +646,44 @@ export default class PreferencesController {
    */
   _subscribeProviderType () {
     this.network.providerStore.subscribe(() => {
-      const { tokens } = this._getTokenRelatedStates()
-      this.store.updateState({ tokens })
+      const { tokens, hiddenTokens } = this._getTokenRelatedStates()
+      this.store.updateState({ tokens, hiddenTokens })
     })
   }
 
   /**
-   * Updates `accountTokens` and `tokens` of current account and network according to it.
+   * Updates `accountTokens`, `tokens`, `accountHiddenTokens` and `hiddenTokens` of current account and network according to it.
    *
    * @param {array} tokens - Array of tokens to be updated.
    *
    */
-  _updateAccountTokens (tokens, assetImages) {
-    const { accountTokens, providerType, selectedAddress } = this._getTokenRelatedStates()
+  _updateAccountTokens (tokens, assetImages, hiddenTokens) {
```

Comment:
Nit: it would be better to document this new parameter in the JSDoc block comment, and `assetImages` (which is also missing)